### PR TITLE
Add softmax support

### PR DESCRIPTION
**Pull request by wayne-chen**
I still have yet to implement the Accelerate implementation due to my main Windows environment. I will try to implement the Accelerate implementation shortly.

### CODE REVIEW

File: bench/layer_creator.hpp [hunk 1]
```diff
@@ -170,6 +170,14 @@ create_layer(const std::string &layer_type, size_t in_size, size_t out_size) {
...
   }
 
+  if (layer_type == "softmax") {
+    auto layer = std::make_unique<RTNeural::SoftmaxActivation<double>>(in_size);
+    return std::move(layer);
+  }
+
+
+
+
```
Comment (jatinchowdhury18) [inline bench/layer_creator.hpp:180]: Please remove these extra newlines.

File: RTNeural/common.h [hunk 2]
```diff
@@ -146,11 +194,35 @@ static inline void sigmoid(const double* in, double* out, size_t dim) noexcept
...
+    vDSP_sveD(out, 1, &exp_sum, dim);
+    vDSP_vsdivD(out, 1, &exp_sum, out, 1, dim);
+}
+
 } // namespace RTNeural
 
 #else // STL backend
 #include <algorithm>
 #include <cmath>
+#include <functional>
```
Comment (jatinchowdhury18) [inline RTNeural/common.h:225]: Do we still need to include `<functional>`?

File: RTNeural/common.h [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 #if defined(USE_EIGEN)
 #include <Eigen/Dense>
+#include <Eigen/LU>
```
Comment (jatinchowdhury18) [inline RTNeural/common.h:5]: Do we still need this include?

File: .github/workflows/bench.yml [hunk 4]
```diff
@@ -83,6 +83,13 @@ jobs:
         ./build/rtneural_layer_bench sigmoid 5 4 4
         ./build/rtneural_layer_bench sigmoid 5 16 16
 
+    - name: Benchmark Softmax
+      run: |
+        ./build/rtneural_layer_bench softmax 5 4 4
+        ./build/rtneural_layer_bench softmax 5 16 16
+
+
+
```
Comment (jatinchowdhury18) [inline .github/workflows/bench.yml:92]: Please remove extra newlines.

File: RTNeural/activation/activation_eigen.h [hunk 5]
```diff
@@ -94,6 +94,36 @@ class SigmoidActivation : public Activation<T>
...
+    }
+
+    SoftmaxActivation(std::initializer_list<size_t> sizes)
+        : SoftmaxActivation(*sizes.begin())
+    {
+    }
+
+    inline void forward(const T* input, T* out) override
+    {
+        inVec = Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic, 1>>(
```
Comment (jatinchowdhury18) [inline RTNeural/activation/activation_eigen.h:115]: After [some commits](https://github.com/jatinchowdhury18/RTNeural/commit/802e85342ae8e01e6f631ed48674929ddda53409#diff-a47ad744e59f0496fa2ccebb9d8a56a36aa4a0747a75966d5c032f2026a9bbdeR27) that I added last weekend, we can actually use `Eigen::Map` with the `Eigen::Aligned16` template argument. You might need to rebase your branch for this to work though.

**Review by jatinchowdhury18 — COMMENTED** [review]
Look pretty good! Just a couple smaller changes that would be good to have before merging. Thanks for adding support for `softmax`!

**Comment by jatinchowdhury18** [comment]
Oh and one more thing: could you also check the box for `softmax` in the README?

File: RTNeural/activation/activation_eigen.h (hunk 5 above)
Comment (wayne-chen) [reply RTNeural/activation/activation_eigen.h]: done!

File: .github/workflows/bench.yml (hunk 4 above)
Comment (wayne-chen) [reply .github/workflows/bench.yml]: done!

File: RTNeural/common.h (hunk 3 above)
Comment (wayne-chen) [reply RTNeural/common.h]: No we don't; I've removed it.

File: RTNeural/common.h (hunk 2 above)
Comment (wayne-chen) [reply RTNeural/common.h]: Removed.

File: bench/layer_creator.hpp (hunk 1 above)
Comment (wayne-chen) [reply bench/layer_creator.hpp]: done!

**Review by jatinchowdhury18 — APPROVED** [review]
Awesome. Looks good to go!

**Pull request merged**